From d35ac84fdcdf287140221f2c03f7749b3810d6a9 Mon Sep 17 00:00:00 2001
From: Alexander Hoffman <alxhoff@gmail.com>
Date: Mon, 7 Apr 2025 14:32:33 +0200
Subject: [PATCH 08/33] [PATCH] kernel: camera: remove csi enent sync for
 recover     CSI enent sync for recover cause kernel null point and cause
 panic.

---
 kernel/nvidia/drivers/media/platform/tegra/camera/csi/csi.c | 6 +++++-
 .../drivers/media/platform/tegra/camera/vi/vi5_fops.c       | 2 ++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/kernel/nvidia/drivers/media/platform/tegra/camera/csi/csi.c b/kernel/nvidia/drivers/media/platform/tegra/camera/csi/csi.c
index d788f7041..8632b9b68 100644
--- a/kernel/nvidia/drivers/media/platform/tegra/camera/csi/csi.c
+++ b/kernel/nvidia/drivers/media/platform/tegra/camera/csi/csi.c
@@ -229,6 +229,7 @@ static int tegra_csi_s_power(struct v4l2_subdev *subdev, int enable)
 	return err;
 }
 
+#if 0
 static int tegra_csi_sync_event(struct v4l2_subdev *subdev,
 	unsigned int sync_events)
 {
@@ -242,6 +243,7 @@ static int tegra_csi_sync_event(struct v4l2_subdev *subdev,
 
 	return err;
 }
+#endif
 
 /*
  * -----------------------------------------------------------------------------
@@ -738,7 +740,9 @@ static struct v4l2_subdev_pad_ops tegra_csi_pad_ops = {
 
 static struct v4l2_subdev_core_ops tegra_csi_core_ops = {
 	.s_power	= tegra_csi_s_power,
-	.sync		= tegra_csi_sync_event,
+#if 0
+    .sync		= tegra_csi_sync_event,
+#endif
 };
 
 static struct v4l2_subdev_ops tegra_csi_ops = {
diff --git a/kernel/nvidia/drivers/media/platform/tegra/camera/vi/vi5_fops.c b/kernel/nvidia/drivers/media/platform/tegra/camera/vi/vi5_fops.c
index fd00d8cd3..6b1f70b50 100644
--- a/kernel/nvidia/drivers/media/platform/tegra/camera/vi/vi5_fops.c
+++ b/kernel/nvidia/drivers/media/platform/tegra/camera/vi/vi5_fops.c
@@ -697,8 +697,10 @@ static int vi5_channel_error_recover(struct tegra_channel *chan,
 		goto done;
 	}
 
+#if 0
 	v4l2_subdev_call(csi_subdev, core, sync,
 		V4L2_SYNC_EVENT_SUBDEV_ERROR_RECOVER);
+#endif
 
 	/* restart vi channel */
 	for (vi_port = 0; vi_port < chan->valid_ports; vi_port++) {
-- 
2.52.0

